src/java/org/apache/cassandra/db/compaction/UnifiedCompactionStrategy.java

```diff
@@ -1094,8 +1095,17 @@ Collection<CompactionAggregate.UnifiedAggregate> getCompactionAggregates(Arena a
                                                                       unbucketed::addAll);
 
             List<CompactionAggregate.UnifiedAggregate> aggregates = new ArrayList<>();
+            Set<CompactionSSTable> selectedSSTables = new HashSet<>();
             for (Bucket bucket : buckets)
-                aggregates.add(bucket.constructAggregate(controller, spaceAvailable, arena));
+            {
+                CompactionAggregate.UnifiedAggregate aggregate = bucket.constructAggregate(controller, spaceAvailable, arena);
+                // if some sstables are already selected, skip the bucket and aggregate
+                if (aggregate.getSelected().sstables().stream().anyMatch(selectedSSTables::contains))
+                    continue;
```

Comment:
With the change to `assignOverlapsInBuckets` this should never happen (indeed, [SonarCloud's coverage report shows it as not visited](https://sonarcloud.io/code?id=cassandra-stargazer&pullRequest=1320&selected=cassandra-stargazer%3Asrc%2Fjava%2Forg%2Fapache%2Fcassandra%2Fdb%2Fcompaction%2FUnifiedCompactionStrategy.java&line=1104)). This part of the change can be reverted now.

(If you want to be extra careful, change the line to throw an exception, run tests to make sure it doesn't fire, and then revert the change.)

Reply:
good point, removed 32820d19594be8ddbf30b0f38c220e1baa1c7fa6. The test will check if there are duplications